optimum/exporters/tflite/convert.py

```diff
@@ -132,10 +137,39 @@ def validate_model_outputs(
         )
 
 
+def create_representative_dataset(signatures, dataset: "Dataset"):
+    def representative_dataset():
+        for sig_name, tf_function in signatures.items():
+            inputs_to_keep = None
+            for example in dataset:
+                if inputs_to_keep is None:
+                    args, kwargs = tf_function.structured_input_signature
+                    args_to_keep = {input_.name for input_ in args if input_.name in example}
+                    kwargs_to_keep = {input_.name for input_ in kwargs.values() if input_.name in example}
+                    inputs_to_keep = args_to_keep | kwargs_to_keep
+                yield sig_name, {name: value for name, value in example.items() if name in inputs_to_keep}
+
+    return representative_dataset
+
+
 def export(
     model: "TFPreTrainedModel",
     config: "TFLiteConfig",
     output: Path,
+    quantization: Optional[Union[str, QuantizationApproach]] = None,
+    fallback_to_float: bool = False,
+    inputs_dtype: Optional[str] = None,
+    outputs_dtype: Optional[str] = None,
+    calibration_dataset_name_or_path: Optional[Union[str, Path]] = None,
+    calibration_dataset_config_name: Optional[str] = None,
+    preprocessor: Optional[Preprocessor] = None,
+    num_calibration_samples: int = 200,
+    calibration_split: Optional[str] = None,
+    primary_key: Optional[str] = None,
+    secondary_key: Optional[str] = None,
+    question_key: Optional[str] = None,
+    context_key: Optional[str] = None,
+    image_key: Optional[str] = None,
```

Comment:
Are all of these arguments related to quantization? Couldn't we separate that into a dataclass, a dict or something? Here some arguments are task specific (question_key, context_key), which should probably be avoided.

Reply:
I can make a dataclass, but I think it will become less readable on what should the function take. 
Disagree on the second part, we support many tasks and need to be able to provide an argument for each of them. This is not problematic to have those arguments living together here.

Reply:
Made it a dataclass.